src/components/form/SelectCondition/index.tsx

```diff
@@ -0,0 +1,93 @@
+import { useDebounceCallback } from '@react-hook/debounce'
+import { SelectConditionModal } from 'components/modals/SelectConditionModal'
+import { Error, ErrorContainer } from 'components/pureStyledComponents/Error'
+import { Row } from 'components/pureStyledComponents/Row'
+import { Textfield } from 'components/pureStyledComponents/Textfield'
+import { TitleControl } from 'components/pureStyledComponents/TitleControl'
+import { TitleValue } from 'components/text/TitleValue'
+import { useConditionContext } from 'contexts/ConditionContext'
+import React, { useCallback } from 'react'
+import styled from 'styled-components'
+
+const InfoMessage = styled.p`
+  color: ${(props) => props.theme.colors.lightGrey};
+  font-size: 14px;
+  font-weight: 400;
+  font-style: italic;
+  line-height: 1.4;
+  margin: 0 0 5px 0;
+  text-align: left;
+
+  &:last-child {
+    margin-bottom: 0;
+  }
+`
+
+export const SelectCondition = ({ onChange }: { onChange?: (id: string) => void }) => {
+  const { condition, errors, loading, setConditionId } = useConditionContext()
+  const [conditionId, setManualConditionId] = React.useState<string>('')
+  const [isModalOpen, setIsModalOpen] = React.useState(false)
+
+  const debouncedHandler = useDebounceCallback((id) => {
+    setConditionId(id)
+  }, 500)
+  const imputHandler = useCallback(
```

Comment:
Please, if you use ̣`React.useState` as in line 29, use in this line `React.useCallback` to keep consistency, also in the lines 43 and 48, and remove the unused import